app/policies/form_policy.rb

```diff
@@ -35,6 +35,7 @@ def can_add_page_routing_conditions?
 private
 
   def users_organisation_owns_form
-    user.organisation_slug == form.org
+    organisation_slug = user.organisation&.slug || user.organisation_slug
```

Comment:
Why are we using safe navigation operator here? I would rather add a guard to the method so only execute the rest of this method if the user has an organisation or organisation_slug

Reply:
The reason we're using the safe navigation operator is because users might not have `organisation_id` set in the table, if they sign in for the first time after this PR has been merged their user record won't be linked to the organisation record.

Reply:
I was thinking something along the lines of adding ` raise Pundit::NotAuthorizedError` at the top of method if the user doesn't have any sort of org assigned to their profile. Users should only really be allowed to view forms if they have an org assigned to them and it matches the forms org.

I would rather handle errors or issues if `user.organisation` is  nil rather than just suppressing them.

Reply:
Doesn't the `resolve` method in the policy already raises an error if the user doesn't have an organisation set?

https://github.com/alphagov/forms-admin/blob/39a6005da4eab03b36e1cecff8649a2e061efc32/app/policies/form_policy.rb#L15

Reply:
`resolve` is part of the policy subclass called `Scope` which is used when you want to return scope the data you are returning back to the user. See https://github.com/alphagov/forms-admin/blob/main/app/controllers/forms_controller.rb#L10 where we scope the forms to just the ones in their org.

🤔  we could maybe move that further up the chain and place it into the different `initialize` methods instead.

Reply:
Nice, That's much better than using the safe navigation operator and shows a user has to have at least an organisation to be able to do anything with the forms.

Reply:
Could we remove the save navigation operator now?